src/key_io.cpp

```diff
@@ -142,7 +142,9 @@ CKey DecodeSecret(const std::string& str)
             key.Set(data.begin() + privkey_prefix.size(), data.begin() + privkey_prefix.size() + 32, compressed);
         }
     }
-    memory_cleanse(data.data(), data.size());
+    if (data.data()) {
```

Comment:
We don't need to `memory_cleanse` if `data.size()` is zero, right?

Reply:
C.f. ae17de5

Reply:
Now calling `memory_cleanse` only if `data.data() != nullptr` and `data.size() > 0`.